api/lib/validate.js

```diff
@@ -166,8 +178,95 @@ function validateOrderbook(orderbook) {
   }
 }
 
+function validateSettings(settings) {
+  if (typeof settings !== 'object') {
+    throw error('Invalid parameter: settings');
```

Comment:
I think there is a case that throwing from async would be considered unconventional in Node, but I'm not sure if that's mostly because most errors in async functions are not immediate.

In the `async.parallel()` case, if the calls are generated programmatically, then invalid requests shouldn't occur and the throws would be like asserts. If the requests are coming from clients, then you could wrap the functions in a small wrapper that catches the exceptions and converts them to callback errors. We could also inline this wrapper into the functions (wrap all the validate calls in a try/catch)